ngraph/core/reference/include/ngraph/runtime/reference/gather.hpp

```diff
@@ -27,149 +30,386 @@ namespace ngraph
+                    }
+                    Strides out_outer_strides(out_ndim, 1);
+                    AxisVector out_outer_axis_order(out_ndim);
+                    std::iota(out_outer_axis_order.begin(), out_outer_axis_order.end(), 0);
+                    CoordinateTransform out_outer_transform(out_shape,
+                                                            out_outer_start_corner,
+                                                            out_outer_end_corner,
+                                                            out_outer_strides,
+                                                            out_outer_axis_order);
+
+                    // Create a CoordinateTransform for "params" that visits the outer "axis"
+                    // dimensions
+                    size_t params_ndim = static_cast<size_t>(params_shape.size());
+                    Coordinate params_outer_start_corner(params_ndim, 0);
+                    Coordinate params_outer_end_corner(params_shape);
+                    for (size_t i = axis; i < params_ndim; i++)
+                    {
+                        params_outer_end_corner[i] = 1;
+                    }
+                    Strides params_outer_strides(params_ndim, 1);
+                    AxisVector params_outer_axis_order(params_ndim);
+                    std::iota(params_outer_axis_order.begin(), params_outer_axis_order.end(), 0);
+                    CoordinateTransform params_outer_transform(params_shape,
+                                                               params_outer_start_corner,
+                                                               params_outer_end_corner,
+                                                               params_outer_strides,
+                                                               params_outer_axis_order);
+
+                    // Create a CoordinateTransform for "indices" that visits only the first element
+                    // along inner most axis
+                    Coordinate indices_outer_start_corner(indices_ndim, 0);
+                    Coordinate indices_outer_end_corner(indices_shape);
+                    if (indices_ndim > 0)
+                    {
+                        indices_outer_end_corner[indices_ndim - 1] = 1;
+                    }
+                    Strides indices_outer_strides(indices_ndim, 1);
+                    AxisVector indices_outer_axis_order(indices_ndim);
+                    std::iota(indices_outer_axis_order.begin(), indices_outer_axis_order.end(), 0);
+                    CoordinateTransform indices_outer_transform(indices_shape,
+                                                                indices_outer_start_corner,
+                                                                indices_outer_end_corner,
+                                                                indices_outer_strides,
+                                                                indices_outer_axis_order);
```

Comment:
Can we avoid using CoordinateTransform?

Reply:
`v0::gather` will be removed when all will be finish.